chainer/variable.py

```diff
@@ -499,38 +499,42 @@ def debug_print(self):
 
         msg = """{summary}
 - device: {device}
-- backend: {background}
+- backend: {backend}
 - shape: {shape}
 - dtype: {dtype}
 - statistics: {stats}
 - grad: {grad}"""
 
         stats_msg = 'mean={0:.8f}, std={1:.8f}'
 
-        try:
-            device = self.data.device
-        except AttributeError:
-            device = 'CPU'
-
-        with cuda.get_device_from_array(self.data) as dev:
+        data = self.data
+        with cuda.get_device_from_array(data) as dev:
             xp = numpy if int(dev) == -1 else cuda.cupy
 
+            if data is None:
+                # `data` can be `None` if constructed without any arguments
+                device = None
+                backend = None
+                stats = None
+            else:
+                device = getattr(data, 'device', 'CPU')
+                backend = type(data)
+                stats = stats_msg.format(float(xp.mean(data)),
+                                         float(xp.std(data)))
+            shape = getattr(data, 'shape', None)
+            dtype = getattr(data, 'dtype', None)
+
             if self.grad is None:
                 grad = None
             elif xp.all(self.grad == 0):
-                grad = 0
+                grad = 0.0
```

Comment:
Is this change related to the bugfix?

Reply:
The motivation was to unify the type with mean and std, but it does not have to be included in this PR as you point out. If this grad needs to be an int, I can change it back.

Reply:
It probably shouldn't be included in this PR anyhow. Reverted the change. Still, please let me know if there is a particular reason to keep it this way.